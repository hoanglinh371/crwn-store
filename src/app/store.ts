import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import cartReducer from '../features/cart/redux/cart.slice';
import categoryReducer from '../features/category/redux/category.slice';
import productReducer from '../features/product/redux/product.slice';

export const store = configureStore({
  reducer: {
    productReducer,
    cartReducer,
    categoryReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
