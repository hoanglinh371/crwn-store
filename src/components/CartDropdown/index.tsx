import React from 'react';
import classNames from 'classnames/bind';
import { useNavigate } from 'react-router-dom';

import { cartItemMock } from 'src/mocks';
import styles from './CartDropdown.module.scss';

import Button from '../Button';
import CartItem from '../CartItem';

const cx = classNames.bind(styles);

const CartDropdown: React.FC = () => {
  const navigate = useNavigate();

  const goToCheckout = () => {
    navigate('/checkout');
  };

  return (
    <div className={cx('container')}>
      <div className={cx('cart-items')}>
        {cartItemMock.map((item) => (
          <CartItem key={item.id} cartItem={item} />
        ))}
      </div>
      <Button type='button' onClick={goToCheckout}>
        CHECKOUT
      </Button>
    </div>
  );
};

export default CartDropdown;
