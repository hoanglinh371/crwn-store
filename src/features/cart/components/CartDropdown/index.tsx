import React from 'react';
import classNames from 'classnames/bind';
import { useNavigate } from 'react-router-dom';
import { useAppSelector } from 'src/app/hooks';

import { selectCartItems } from '../../redux/cart.selectors';
import styles from './CartDropdown.module.scss';

import Button from 'src/components/Button';
import CartItem from '../CartItem';

const cx = classNames.bind(styles);

const CartDropdown: React.FC = () => {
  const navigate = useNavigate();
  const cartItems = useAppSelector(selectCartItems);

  const goToCheckout = () => {
    navigate('/checkout');
  };

  return (
    <div className={cx('container')}>
      <div className={cx('cart-items')}>
        {cartItems.map((item) => (
          <CartItem key={item.id} cartItem={item} />
        ))}
      </div>
      <Button type='button' onClick={goToCheckout}>
        CHECKOUT
      </Button>
    </div>
  );
};

export default CartDropdown;
