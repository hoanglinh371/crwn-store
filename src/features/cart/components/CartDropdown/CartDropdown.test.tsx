import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from 'src/app/store';
import CartDropdown from '.';

describe('CartDropdown component', () => {
  afterEach(cleanup);

  it('Should_MatchSnapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CartDropdown />
        </BrowserRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
