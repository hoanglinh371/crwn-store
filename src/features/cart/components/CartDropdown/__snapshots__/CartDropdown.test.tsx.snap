// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown component Should_MatchSnapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="cart-items"
    />
    <button
      aria-label="button"
      class="container"
      type="button"
    >
      CHECKOUT
    </button>
  </div>
</div>
`;
